jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java

```diff
@@ -0,0 +1,11 @@
+public class Main {
+
+  public void constantIndexOf() {
+    String s1 = "abcabc";
+    Character b = 'b';
+    assert s1.indexOf(b) == 1;
+    assert s1.indexOf(b, -10) == 1;
+    assert s1.indexOf(b, 3) == 4;
+    assert s1.indexOf(b, 10) == -1;
```

Comment:
In cbmc, it's generally not enough to check for assertion success. After all, cbmc's job is to violate assertions, so if there's nothing to violate, we can't tell if cbmc did anything at all.
I recommend looking at the tests in https://github.com/diffblue/cbmc/tree/develop/jbmc/regression/jbmc-strings/CompareToConstantEvaluation that exercise failing assertions.

Reply:
Thanks for the great advice and reference. Indeed, I had a test.desc for this test that does not use models-library, which is verification is expected to be failed. Is it better to add the test? The detailed verification failure is https://diffblue.atlassian.net/browse/TG-8284?focusedCommentId=23418&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-23418

Reply:
That test you link doesn't look like one you would want to add. It fails on a NullPointerException, which is not what you're looking for. Happy to discuss in person.

Reply:
@allredj , I added a test to check assertion failure. Could you re-review?